in_toto/verifylib.py

```diff
@@ -1043,8 +1044,105 @@ def reduce_chain_links(chain_link_dict):
 
   return reduced_chain_link_dict
 
+def verify_delegations(layout, chain_link_dict):
+  """
+  <Purpose>
+    Checks if any step has been delegated by the functionary, recurses into
+    the delegation and replaces the layout object in the chain_link_dict
+    by an equivalent link object
+
+  <Arguments>
+    layout:
+            The layout specified by the project owner against which the
+            threshold will be verified.
+
+    chain_link_dict:
+            A dictionary of key-link pair with step names as keys. For each
+            step name, there are one or more keyids and corresponding
+            link objects.
+
+  <Exceptions>
+    raises an Exception if verification of the delegated step fails.
```

Comment:
Me neither. I think it's fine like this. One thing that bugs me more is the inconsistent usage of  the `<Exception>` in function docstrings. Sometimes we only list exceptions that are actually raised in the very function, and sometimes we also raise exceptions that get raised in called functions (like here). I'm not sure which one is better.